src/rdo.rs

```diff
@@ -967,17 +967,17 @@ pub fn rdo_partition_decision(
   let mut best_rd = cached_block.rd_cost;
   let mut best_pred_modes = cached_block.part_modes.clone();
 
+  let cw_checkpoint = cw.checkpoint();
+  let w_pre_checkpoint = w_pre_cdef.checkpoint();
+  let w_post_checkpoint = w_post_cdef.checkpoint();
+
   for &partition in partition_types {
     // Do not re-encode results we already have
     if partition == cached_block.part_type {
       continue;
     }
-
     let mut cost: f64 = 0.0;
     let mut child_modes = std::vec::Vec::new();
-    let cw_checkpoint = cw.checkpoint();
-    let w_pre_checkpoint = w_pre_cdef.checkpoint();
-    let w_post_checkpoint = w_post_cdef.checkpoint();
 
     match partition {
       PartitionType::PARTITION_NONE => {
```

Comment:
For your understanding, this case is only hit for current largest block to start with top-down search, i.e. 64x64.
Then, 32x32, 16x16, 8x8 has no chance to see this case of PARTITION_NONE since it has been already tested by line 1052 below when its parent block is split.